pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/EqualsPredicateEvaluatorFactory.java

```diff
@@ -82,202 +82,6 @@ public static BaseRawValueBasedPredicateEvaluator newRawValueBasedEvaluator(EqPr
     }
   }
 
-  private static final class DictionaryBasedEqPredicateEvaluator extends BaseDictionaryBasedPredicateEvaluator {
```

Comment:
Suggest keeping them as inner class and changing them to `public` access (same as range predicate). All of them are under the scope of EQ predicate

Reply:
I mentioned why I made this change here https://github.com/apache/pinot/pull/10043#discussion_r1063363939
I feel the single interface approach made this less cumbersome; I wouldn't need to care about how long the class names are for formatting reasons if the level of coupling were reduced.

Reply:
I have instead introduced the `traits` package to allow extraction of values/ranges from predicate evaluators without needing to know the concrete type. This reduces coupling and improves local readability within the operator.